pkg/controller/direct/alloydb/mapping.go

```diff
@@ -0,0 +1,231 @@
+	if in == nil {
+		return nil
+	}
+	out := &api.WeeklySchedule{
+		DaysOfWeek: in.DaysOfWeek,
+		StartTimes: StartTimes_KRMToApi(ctx, in.StartTimes),
+	}
+
+	return out
+}
+
+func StartTimes_KRMToApi(ctx *MapContext, in []krm.ClusterStartTimes) []*api.GoogleTypeTimeOfDay {
+	out := make([]*api.GoogleTypeTimeOfDay, len(in))
+	for i, v := range in {
+		out[i] = Time_KRMToApi(ctx, LazyPtr(v))
+	}
+	return out
+}
+
+func Time_KRMToApi(ctx *MapContext, in *krm.ClusterStartTimes) *api.GoogleTypeTimeOfDay {
+	if in == nil {
+		return nil
+	}
+	out := &api.GoogleTypeTimeOfDay{
+		Hours:   int64(ValueOf(in.Hours)),
+		Minutes: int64(ValueOf(in.Minutes)),
+		Nanos:   int64(ValueOf(in.Nanos)),
+		Seconds: int64(ValueOf(in.Seconds)),
+	}
+	return out
+}
+
+func EncryptionConfig_KRMToApi(ctx *MapContext, in *krm.ClusterEncryptionConfig) *api.EncryptionConfig {
+	if in == nil {
+		return nil
+	}
+	out := &api.EncryptionConfig{
+		KmsKeyName: in.KmsKeyNameRef.External,
+	}
+	return out
+}
+
+func TimeBasedRetention_KRMToApi(ctx *MapContext, in *krm.ClusterTimeBasedRetention) *api.TimeBasedRetention {
+	if in == nil {
+		return nil
+	}
+	out := &api.TimeBasedRetention{
+		RetentionPeriod: ValueOf(in.RetentionPeriod),
+	}
+
+	return out
+}
+
+func QuantityBasedRetention_KRMToApi(ctx *MapContext, in *krm.ClusterQuantityBasedRetention) *api.QuantityBasedRetention {
+	if in == nil {
+		return nil
+	}
+	out := &api.QuantityBasedRetention{
+		Count: int64(ValueOf(in.Count)),
+	}
+
+	return out
+}
+
+func ContinuousBackupConfig_KRMToApi(ctx *MapContext, in *krm.ClusterContinuousBackupConfig) *api.ContinuousBackupConfig {
+	if in == nil {
+		return nil
+	}
+	out := &api.ContinuousBackupConfig{
+		Enabled:            ValueOf(in.Enabled),
+		EncryptionConfig:   EncryptionConfig_KRMToApi(ctx, in.EncryptionConfig),
+		RecoveryWindowDays: int64(ValueOf(in.RecoveryWindowDays)),
+	}
+
+	return out
+}
+
+func InitialUser_KRMToApi(ctx *MapContext, in *krm.ClusterInitialUser) *api.UserPassword {
+	if in == nil {
+		return nil
+	}
+	out := &api.UserPassword{
+		User: ValueOf(in.User),
+	}
+	if in.Password.Value != nil {
+		out.Password = ValueOf(in.Password.Value)
+	} else {
+		out.Password = in.Password.ValueFrom.SecretKeyRef.Key
+	}
+	return out
+}
+
+func ClusterStatusFromApi(ctx *MapContext, in *api.Cluster) *krm.AlloyDBClusterStatus {
+	if in == nil {
+		return nil
+	}
+	out := &krm.AlloyDBClusterStatus{
+		BackupSource:         BackupSource_KRMFromApi(ctx, in.BackupSource),
+		ContinuousBackupInfo: ContinuousBackupInfo_KRMFromApi(ctx, in.ContinuousBackupInfo),
+		DatabaseVersion:      LazyPtr(in.DatabaseVersion),
+		EncryptionInfo:       EncryptionInfo_KRMFromApi(ctx, in.EncryptionInfo),
+		MigrationSource:      MigrationSource_KRMFromApi(ctx, in.MigrationSource),
+		Name:                 LazyPtr(in.Name),
+		Uid:                  LazyPtr(in.Uid),
+	}
+	return out
+}
+
+func BackupSource_KRMFromApi(ctx *MapContext, in *api.BackupSource) []krm.ClusterBackupSourceStatus {
```

Comment:
Ouch ... do we have the cardinality wrong in our API?  Still, at least it looks like we have it wrong the less-problematic way.